src/plan/markcompact/global.rs

```diff
@@ -218,6 +238,27 @@ impl<VM: VMBinding> MarkCompact<VM> {
 
         res
     }
+
+    pub fn work_bucket_stage_config() -> WorkBucketStageConfig {
+        let mut config = WorkBucketStageConfig::default();
+        // Add cusotm stages before release
+        config.bulk_insert_before(
+            vec![
+                // Calculate forwarding addresses for live objects
+                CALCULATE_FORWARDING_STAGE,
+                // Second round transitive closure
+                SECOND_ROOTS_STAGE,
+                // Forwarding weak/final references
+                REF_FORWARDING_STAGE,
+                FINALIZABLE_FORWARDING_STAGE,
+                VM_REF_FORARDING_STAGE,
+                // Compaction
+                COMPACT_STAGE,
+            ],
```

Comment:
If we have a `MarkCompactStage` that implements `enum_map::Enum`, we can convert all of its variants into `WorkBucketStage` automatically, such as: `            (0..MarkCompactStage::LENGTH).map(|i|MarkCompactStage::from_usize(i).as_stage()).collect_vec()
`